src/AggregateFunctions/AggregateFunctionGroupArraySorted.cpp

```diff
@@ -0,0 +1,394 @@
+    }
+
+    void serialize(ConstAggregateDataPtr __restrict place, WriteBuffer & buf, std::optional<size_t> /* version */) const override
+    {
+        auto & values = this->data(place).values;
+        size_t size = values.size();
+        writeVarUInt(size, buf);
+
+        if constexpr (std::is_same_v<T, Field>)
+        {
+            for (const Field & element : values)
+            {
+                if (element.isNull())
+                {
+                    writeBinary(false, buf);
+                }
+                else
+                {
+                    writeBinary(true, buf);
+                    serialization->serializeBinary(element, buf, {});
+                }
+            }
+        }
+        else
+        {
+            for (const auto & element : values)
+                writeBinaryLittleEndian(element, buf);
+        }
+    }
+
+    void deserialize(AggregateDataPtr __restrict place, ReadBuffer & buf, std::optional<size_t> /* version */, Arena * arena) const override
+    {
+        size_t size = 0;
+        readVarUInt(size, buf);
+
+        if (unlikely(size > max_elements))
+            throw Exception(ErrorCodes::TOO_LARGE_ARRAY_SIZE, "Too large array size, it should not exceed {}", max_elements);
+
+        auto & values = this->data(place).values;
+        values.resize(size, arena);
+
+        if constexpr (std::is_same_v<T, Field>)
+        {
+            for (Field & element : values)
+            {
+                UInt8 is_null = 0;
+                readBinary(is_null, buf);
+                if (!is_null)
+                    serialization->deserializeBinary(element, buf, {});
+            }
+        }
+        else
+        {
+            for (auto & element : values)
+                readBinaryLittleEndian(element, buf);
+        }
+    }
+
+    void insertResultInto(AggregateDataPtr __restrict place, IColumn & to, Arena * arena) const override
+    {
+        this->data(place).insertResultInto(to, max_elements, arena);
+    }
+
+    bool allocatesMemoryInArena() const override { return true; }
+
+private:
+    UInt64 max_elements;
+    SerializationPtr serialization;
+};
+
+template <typename T>
+using GroupArraySortedHeap = GroupArraySorted<GroupArraySortedDataHeap<T>, T>;
+
+template <typename T>
+using GroupArraySortedSort = GroupArraySorted<GroupArraySortedDataSort<T>, T>;
+
+template <template <typename> class AggregateFunctionTemplate, typename ... TArgs>
+AggregateFunctionPtr createWithNumericOrTimeType(const IDataType & argument_type, TArgs && ... args)
+{
+    WhichDataType which(argument_type);
+
+    if (which.idx == TypeIndex::Date) return std::make_shared<AggregateFunctionTemplate<UInt16>>(std::forward<TArgs>(args)...);
+    if (which.idx == TypeIndex::DateTime) return std::make_shared<AggregateFunctionTemplate<UInt32>>(std::forward<TArgs>(args)...);
+    if (which.idx == TypeIndex::IPv4) return std::make_shared<AggregateFunctionTemplate<IPv4>>(std::forward<TArgs>(args)...);
```

Comment:
This code is part of `groupArray` aggregation function.
Field is ok, for any complex type. Comparison of Field also need to work for any non complex type, the same as comparison of column for those non complex type. Because we can create `Tuple` with non complex type, and sorting result must be the same as for just non complex type.
For Date32 we can use Int32.
For DecimalN we can also use ColumnDecimal specializations.